File: crypto_dashboard_client/lib/models/crypto_pair.dart
class CryptoPair {
  final String pairName;
  final DateTime date;
  final double open;
  final double close;
  final double high;
  final double low;
  final double price;
  final int volume;
  final double changePercentage;

  CryptoPair({
    required this.pairName,
    required this.date,
    required this.open,
    required this.close,
    required this.high,
    required this.low,
    required this.price,
    required this.volume,
    required this.changePercentage,
  });

  factory CryptoPair.fromJson(Map<String, dynamic> json) {
    return CryptoPair(
      pairName: json['pairName'],
      date: DateTime.parse(json['date']),
      open: json['open'].toDouble(),
      close: json['close'].toDouble(),
      high: json['high'].toDouble(),
      low: json['low'].toDouble(),
      price: json['price'].toDouble(),
      volume: json['volume'],
      changePercentage: json['changePercentage'].toDouble(),
    );
  }
}

File: CryptoDashboard.Api/Models/CryptoPair.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CryptoDashboard.Api.Models
{
    [Table("cryptopairs")]
    public class CryptoPair
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [Column("pairname")]
        [MaxLength(10)]
        public string PairName { get; set; }

        [Required]
        [Column("date")]
        public DateTime Date { get; set; }

        [Required]
        [Column("price")]
        public decimal Price { get; set; }

        [Required]
        [Column("open")]
        public decimal Open { get; set; }

        [Required]
        [Column("high")]
        public decimal High { get; set; }

        [Required]
        [Column("low")]
        public decimal Low { get; set; }

        [Required]
        [Column("volume")]
        public long Volume { get; set; }

        [Required]
        [Column("changepercentage")]
        public decimal ChangePercentage { get; set; }
    }

    public class PairAverageDifference
    {
        public string PairName { get; set; }
        public double AvgDiff { get; set; }
        public double PercentageDiff { get; set; }
    }


    public class LowAndHigh
    {
        public decimal LowestPrice { get; set; }
        public decimal HighestPrice { get; set; }
    }

    public class LogEntry
    {
        public DateTime BeginDate { get; set; }
        public DateTime EndDate { get; set; }
        public DateTime RequestDate { get; set; }
    }

    public class ErrorLogEntry
    {
        public string Action { get; set; }
        public string ErrorMessage { get; set; }
        public string StackTrace { get; set; }
        public DateTime Date { get; set; }
    }

}


